src/lib/storage/table.cpp

```diff
@@ -1,84 +1,111 @@
+  _column_names.emplace_back(name);
+  _column_types.emplace_back(type);
+  _column_nullable.emplace_back(nullable);
 }
 
 void Table::add_column(const std::string& name, const std::string& type, const bool nullable) {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  DebugAssert(_chunks[0]->size() == 0, "Adding columns is only allowed if the table does not have any entries.");
+  add_column_definition(name, type, nullable);
+  resolve_data_type(type, [&](const auto data_type_t) {
+    using ColumnDataType = typename decltype(data_type_t)::type;
+    const auto value_segment = std::make_shared<ValueSegment<ColumnDataType>>(nullable);
+    _chunks[0]->add_segment(value_segment);
+  });
 }
 
 void Table::create_new_chunk() {
-  // Implementation goes here
+  _chunks.emplace_back(std::make_shared<Chunk>());
+
+  for (unsigned int i = 0; i < _column_types.size(); i++) {
+    resolve_data_type(_column_types[i], [&](const auto data_type_t) {
+      using ColumnDataType = typename decltype(data_type_t)::type;
+      const auto value_segment = std::make_shared<ValueSegment<ColumnDataType>>(_column_nullable[i]);
+      _chunks.back()->add_segment(value_segment);
+    });
+  }
 }
 
 void Table::append(const std::vector<AllTypeVariant>& values) {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  if (_chunks.back()->size() == _target_chunk_size) {
+    create_new_chunk();
+  }
+  _chunks.back()->append(values);
 }
 
 ColumnCount Table::column_count() const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  return static_cast<ColumnCount>(_column_names.size());
 }
 
 uint64_t Table::row_count() const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  return (_chunks.size() - 1) * _target_chunk_size + _chunks.back()->size();
 }
 
 ChunkID Table::chunk_count() const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  return static_cast<ChunkID>(_chunks.size());
 }
 
 ColumnID Table::column_id_by_name(const std::string& column_name) const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  for (unsigned int i = 0; i < _column_names.size(); i++) {
+    if (_column_names[i] == column_name) {
+      return static_cast<ColumnID>(i);
+    }
+  }
+  throw std::logic_error("Table does not contain column with the requested name.");
 }
 
 ChunkOffset Table::target_chunk_size() const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  return _target_chunk_size;
 }
 
 const std::vector<std::string>& Table::column_names() const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  return _column_names;
 }
 
 const std::string& Table::column_name(const ColumnID column_id) const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  if (column_id > _column_names.size()) {
+    throw std::logic_error("Table does not contain column with the requested id.");
+  }
```

Comment:
```suggestion
    DebugAssert(column_id < _column_names.size(), "Table does not contain column with the requested id.");
```

Reply:
The tests expect a logic_error to be thrown here, otherwise I would agree.